Introduce commands to generate tests for function at cursor

krobelus opened the pull request.

krobelus committed:
Add symflower user mode, for quick access to everyday commands

I chose u for switching to the *symflower* buffer (often used for
debugging) to follow the popular kakoune-buffers plugin that maps u
to switch to the *debug* buffer.

krobelus committed:
Embelish the documentation with Asciinema screencasts

Add three screencasts that showcase three different features:

1. unit test generation
2. user mode (shortcuts)
3. unit test skeleton generation

add them next to their respective documentation entries.  Except for
the first one, because we use that as opener.  It seems stronger than
the static image of Copy.java, let's just move that to "Quick Start"
(which fits because it uses the same Copy.java).

krobelus committed:
Introduce commands to generate tests for the current function

This uses the directory of the current file as workspace, which
improves performance when the project has many files.

Soon enough, this will fail for Java (because imports require a sane
classpath), but we can worry about that later.

zimmski commented:
Usage of the README needs to be adapted, since we now allow unit test skeletons as well

zimmski commented on README.md:
```diff
@@ -27,9 +27,12 @@ This plugin integrates [Symflower CLI](https://symflower.com/en/products/symflow
 
 - The `symflower` command generates tests for source files in the current directory.
 - The `symflower-alternate-file` command jumps between a source file and the generated test file.
+- The `symflower-buffile` command generates tests for the current buffer's file.
```
Usage of the README needs to be adapted, since we now allow unit test skeletons as well

zimmski commented on rc/symflower.kak:
```diff
@@ -49,3 +75,10 @@ define-command symflower-alternative-file -docstring 'Jump to the alternate file
 }}
+
+declare-user-mode symflower
+map global symflower a %{: symflower-alternative-file<ret>} -docstring "implementation ←→ Symflower test"
```
Can you please add a feature section to the README to showcase both the unit test generation and unit test skeleton adding with a GIF and text for the user on what these features are and how to use them

krobelus replied:
Added screencasts but no separate feature section yet

krobelus committed:
Remove symflower-enable and symflower-disable

The advent of more fine-grained symbol selection in the next patch
(which uses the symbol at cursor) puts symflower-enable in a weird
spot. Generating everything is inconsistent with the rest of the
command palette.  Adapting to also use the cursor position seems out
of place for a write hook.

krobelus committed:
fix, Typo

krobelus committed:
Move "Usage" section after "Quick Start"

The next patches will add more content to usage, but it's not really
needed for the quick start example, so let's make that even quicker.
